packages/frontendmu-nuxt/components/home/SponsorCarousel.vue

```diff
@@ -0,0 +1,150 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+import { useSponsorStore } from '@/store/sponsorStore'
+
+const sponsorStore = useSponsorStore()
+// Get all sponsors, sorted by their most recent event date descending
+const sortedSponsors = computed(() => {
+  return [...sponsorStore.sponsors]
+    .sort((a, b) => {
+      // Find most recent meetup date for each sponsor
+      const aLatest = a.meetups.reduce((max, m) => m.date > max ? m.date : max, '')
+      const bLatest = b.meetups.reduce((max, m) => m.date > max ? m.date : max, '')
+      return bLatest.localeCompare(aLatest)
+    })
+})
+
+function sponsorLogoUrl(sponsor: any) {
+  if (typeof sponsor.logo === 'string') {
+    return `https://directus.frontend.mu/assets/${sponsor.logo}`
+  }
+  return ''
+}
+
+const carouselRef = ref<HTMLDivElement | null>(null)
```

Comment:
Could try `templateRefs` when we reach Vue 3.5+ https://vuejs.org/guide/essentials/template-refs